lib/index.js

```diff
@@ -213,7 +213,9 @@ Segment.prototype.normalize = function(msg) {
       ctx.traits.crossDomainId = crossDomainId;
     }
   }
-  if (query) ctx.campaign = utm(query);
+  if (query && !ctx.campaign) {
+    ctx.campaign = utm(query);
```

Comment:
A possible improvement might be to do a merge if `ctx.campaign` exists. e.g. maybe you track referrer yourself, but still want the other context.campaign fields to be filled by us.